### ISSUE TITLE
Unknown response frame type

### ISSUE THREAD

**Issue by sandy-shirt** [jasonrbriggs/stomp.py#239]
this is my first time use stomp 
there is some message i want to get from an ip
but the message 'Unknown response frame type' occur 
i dont know whats the problem

**Comment by chuckchan24** [comment]
I also encountered this issue these days. 
I want to receive some message from my activeMQ server, 
and the message was sent by Java code, using OpenWire protocol.

**Comment by sandy-shirt** [comment]
> 
> 
> I also encountered this issue these days.
> I want to receive some message from my activeMQ server,
> and the message was sent by Java code, using OpenWire protocol.

I have fix this problem  by
change the port 61616 into 61613 
I don't know whether it is suit for you

**Comment by chuckchan24** [comment]
> > I also encountered this issue these days.
> > I want to receive some message from my activeMQ server,
> > and the message was sent by Java code, using OpenWire protocol.
> 
> I have fix this problem by
> change the port 61616 into 61613
> I don't know whether it is suit for you

Yes, I am using the port 61616. 
but there are other projects using this port.
It 's hard to change it on all projects. 
Is there another way to fix it?

**Comment by jasonrbriggs** [comment]
You are trying to connect stomp.py to the default port ActiveMQ uses for connections -- not to the port used for stomp protocol connections.  That's why you get an invalid frame -- the server is sending data that stomp.py has no idea how to interpret.  You need to use the correct stomp port - I suggest reading this first: https://activemq.apache.org/stomp.html

**Comment by chuckchan24** [comment]
Problem sovled! Thank you!!

**Comment by Prathik-Jain** [comment]
It could also be that you're connecting to STOMP+SSL and not setting the SSL 

see https://github.com/jasonrbriggs/stomp.py/issues/172

**Comment by juocal** [comment]
> You are trying to connect stomp.py to the default port ActiveMQ uses for connections -- not to the port used for stomp protocol connections. That's why you get an invalid frame -- the server is sending data that stomp.py has no idea how to interpret. You need to use the correct stomp port - I suggest reading this first: https://activemq.apache.org/stomp.html

@jasonrbriggs Thank you for your comment! This also solved my issue!

I tried to connect stomp.py to **port 61616** which I am using in my **Java application**. This did not work correctly. I was able to "connect" without authentification (although it was configured) but was only getting wierdly formatted data (as stated by OP) which was logged with "Unknown response frame type' occur" in the stdout of my python application.

After reading your comment, I changed the **port to 61613** in my **python application** with **stomp.py** and was able to connect sucessfully and receive the data. Even the authentification worked then :)

I hope I might help others finding this issue via google as I did!
